src/samples/audio/nxp_detect_test.c

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright 2021 NXP
+//
+// Author: Cristina Feies <cristina.ilie@nxp.com>
+// Author: Viorel Suman <viorel.suman@nxp.com>
+
+#include <sof/samples/audio/nxp_detect_test.h>
+
+static __aligned(64) uint8_t preprocessed_data[CONFIG_PREPROCESSED_SIZE];
+static __aligned(8) uint8_t confidences[4];
+
+//#define PRINT
+
+int postprocess(uint8_t confidences[4])
+{
+	int i;
+	uint8_t max_confidence = confidences[0];
+	int max_idx = 0;
+
+	for (i = 1; i < 4; i++) {
+		if (max_confidence < confidences[i]) {
+			max_confidence = confidences[i];
+			max_idx = i;
+		}
+	}
+	if (max_confidence < 128)
+		max_idx = 1;
+	return max_idx;
+}
+
+void nxp_detect_test(struct comp_dev *dev,
+				const struct audio_stream *source,
+				uint32_t frames)
+{
+	void *src;
+	uint32_t count = frames; /**< Assuming single channel */
+	uint32_t sample;
+	uint32_t cycles_per_frame; /**< Clock cycles required per frame */
+	uint32_t one_sec_samples = KEY_LEN;
+	uint32_t half_sec_samples = one_sec_samples / N;
+
+	/* synthetic load */
+	if (test_keyword_get_load_mips(dev)) {
+		/* assuming count is a processing frame size in samples */
+		cycles_per_frame = (test_keyword_get_load_mips(dev) * 1000000 * count)
+				/ source->rate;
+		idelay(cycles_per_frame);
+	}
+	/* perform detection within current period */
+	for (sample = 0; sample < count && !test_keyword_get_detected(dev); ++sample) {
+		src = (test_keyword_get_sample_valid_bytes(dev) * 8 == 16U) ?
+			audio_stream_read_frag_s16(source, sample) :
+			audio_stream_read_frag_s32(source, sample);
+		if (test_keyword_get_input_size(dev) < IN_BUFF_SIZE) {
+			if (test_keyword_get_sample_valid_bytes(dev) == 16U)
+				test_keyword_set_input_elem(dev,
+					test_keyword_get_input_size(dev),
+					*((int16_t *)src));
+			else
+				test_keyword_set_input_elem(dev,
+					test_keyword_get_input_size(dev),
+					*((int32_t *)src));
+			test_keyword_set_input_size(dev, test_keyword_get_input_size(dev) + 1);
+		}
+		if (test_keyword_get_input_size(dev) > one_sec_samples) {
+			uint64_t time_start;
+			uint64_t time_stop;
+			struct timer *timer = timer_get();
+			int result;
+			int i, j;
+
+#ifdef PRINT
```

Comment:
Is this need it?
If you want the message below to be shown only on debug use `comp_dbg()`.